superset/views/utils.py

```diff
@@ -99,23 +99,28 @@ def get_viz(
     return viz_obj
 
 
-def get_form_data(slice_id=None, use_slice_data=False):
+def get_form_data(
+    slice_id: Optional[int] = None, use_slice_data: bool = False
+) -> Tuple[Dict[str, Any], Optional[Slice]]:
     form_data = {}
-    post_data = request.form.get("form_data")
+    request_form_data = request.form.get("form_data")
     request_args_data = request.args.get("form_data")
-    # Supporting POST
-    if post_data:
-        form_data.update(json.loads(post_data))
-    # request params can overwrite post body
+    if request_form_data:
+        form_data.update(json.loads(request_form_data))
+    # request params can overwrite the body
     if request_args_data:
         form_data.update(json.loads(request_args_data))
 
+    # Fallback to using the Flask globals (used for cache warmup) if defined.
+    if not form_data and hasattr(g, "form_data"):
+        form_data = getattr(g, "form_data")
+
     url_id = request.args.get("r")
     if url_id:
         saved_url = db.session.query(models.Url).filter_by(id=url_id).first()
         if saved_url:
             url_str = parse.unquote_plus(
-                saved_url.url.split("?")[1][10:], encoding="utf-8", errors=None
+                saved_url.url.split("?")[1][10:], encoding="utf-8"
```

Comment:
@villebro `unquote_plus` was raising a typing issue as it expected `errors` to be of type `str`. Per [here](https://github.com/python/cpython/blob/master/Lib/urllib/parse.py#L653) it seems if `errors is None` it gets set to `replace` which is the default, thus there's no need to specify it.